audio/libsndfile/Portfile

```diff
@@ -1,47 +1,55 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
-PortGroup           cmake 1.0
 PortGroup           github 1.0
 
 github.setup        libsndfile libsndfile 1.0.31
-revision            0
+revision            1
+
 categories          audio
 maintainers         nomaintainer
 platforms           darwin
 
 description         read and write files containing sampled sound
 long_description    \
-	libsndfile is a C library for reading and writing files	\
-	containing sampled sound through one standard library interface. \
-	libsndfile has the following main features: \
-	ability to read and write a large number of file formats, \
-	a simple, elegant and easy to use API, \
-	and on the fly format conversion.
+    libsndfile is a C library for reading and writing files \
+    containing sampled sound through one standard library interface. \
+    libsndfile has the following main features: \
+    ability to read and write a large number of file formats, \
+    a simple, elegant and easy to use API, \
+    and on the fly format conversion.
 license             LGPL-2.1+
+homepage            https://libsndfile.github.io/libsndfile/
+
 checksums           \
-      rmd160  8663673c3f8eaf463c7b499c732c53c9bba03c04 \
-      sha256  f082718e84a27d99c0e349151344b71dc2032e7aa2d799af0da48a8a270aa06a \
-      size    662676
-depends_build-append  port:pkgconfig
-depends_lib           port:flac port:libogg port:libvorbis
+    rmd160  ae3fc5bbcb10a034f3edc1240acacd9f1ec349a7 \
+    sha256  a8cfb1c09ea6e90eff4ca87322d4168cdbe5035cb48717b40bf77e751cc02163 \
+    size    875335
+
+github.tarball_from releases
+master_sites        https://github.com/libsndfile/libsndfile/releases/download/${version}
+distfiles           libsndfile-${version}.tar.bz2
 
-patchfiles            patch-CMakeLists.txt.diff
+depends_build       port:pkgconfig
+depends_lib         port:flac port:libogg port:libvorbis port:libopus
 
-configure.args-append -DBUILD_REGTEST=OFF \
-                      -DBUILD_SHARED_LIBS=ON \
-                      -DENABLE_EXTERNAL_LIBS=ON \
-                      -DENABLE_EXPERIMENTAL=OFF
+configure.args \
+    --disable-alsa \
+    --disable-octave \
+    --disable-silent-rules \
+    --disable-sqlite \
+    --disable-test-coverage \
+    --enable-external-libs
 
 variant no_external_libs description {Disable FLAC, Ogg and Vorbis support} {
     depends_build-delete  port:pkgconfig
-    depends_lib-delete    port:flac port:libogg port:libvorbis
-    configure.args-append -DENABLE_EXTERNAL_LIBS=OFF
-    configure.args-delete -DENABLE_EXTERNAL_LIBS=ON
+    depends_lib-delete    port:flac port:libogg port:libvorbis port:libopus
+    configure.args-append --disable-external-libs
+    configure.args-delete --enable-external-libs
 }
 
-variant experimental description {Enable experimental support for Opus, Speex} {
```

Comment:
what's the deal with this variant - why did you remove this? Does the new version not have this option anymore and is it included now by default (in that case the dependencies are probably still needed) or is the support just removed completely?

Reply:
I pointlessly added it for one release. Do I need to keep it in case someone just added it to their workflow?

It otherwise automatically detects FLAC, Vorbis, Opus, and even Speex, regardless of the alleged "experimental" status of the latter two.

Reply:
automatically detection is probably not really what we want as then the end result will depend on whether the user has these other ports installed or not (i.e., not a reproducible build). So if you want to enable the support unconditionally for all user you should add the required dependencies, otherwise the port should be build without the support by default and one can enable it using variants.

Reply:
I've re-added the experimental thing, clarifying what it does this time. Indeed, some of the documentation claimed that Opus was gated behind "experimental", but that's no longer the case. The only thing that appears to still be gated behind `--enable-experimental` now is Speex support. It will still detect Speex, but it won't link in support for it without that flag. If you can suggest a clean way to make sure that `no_external_libs` and `experimental` are mutually exclusive, I'm all ears. Considering it should still build if both are specified, except that the `experimental` variant will do absolutely nothing useful other than adding an unused dependency, since the `--disable-external-libs` switch should disable that effectively anyway.

Reply:
You could probably do something like:
```variant experimental conflicts no_external_libs description {Enable experimental Speex support} {```
and add a conflict the other way around as well